components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -809,14 +809,16 @@ subroutine ocn_diagnostic_solve_kineticEnergy(normalVelocity, &
       !$omp do schedule(runtime) private(i, iEdge, r_tmp, k)
 #endif
       do iVertex = 1, nVertices
-        kineticEnergyVertex(:, iVertex) = 0.0_RKIND
+         kmin = minLevelVertexBot(iVertex)
+         kmax = maxLevelVertexTop(iVertex)
+         kineticEnergyVertex(:, iVertex) = 0.0_RKIND
         do i = 1, vertexDegree
           iEdge = edgesOnVertex(i, iVertex)
           r_tmp = dcEdge(iEdge) * dvEdge(iEdge) * 0.25_RKIND / areaTriangle(iVertex)
 #ifdef MPAS_OPENACC
           !$acc loop seq
 #endif
-          do k = 1, nVertLevels
+          do k = kmin, kmax
```

Comment:
Hi @philipwjones ! We changed the bounds of the loop. Are you happy with the OpenACC call to be left as it was before?

Reply:
Yes, that's fine. Loop extent won't have an impact. Thanks for asking.